x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.test.ts

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  fetchLogstashStats,
+  fetchLogstashState,
+  processStatsResults,
+  processLogstashStateResults,
+} from './get_logstash_stats';
+import sinon from 'sinon';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const logstashStatsResultSet = require('./__mocks__/fixtures/logstash_stats_results');
+
+const resultsMap = new Map();
+
+// Load data for state results.
+['1n1p', '1nmp', 'mnmp'].forEach((data) => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  resultsMap.set(data, require(`./__mocks__/fixtures/logstash_state_results_${data}`));
+});
+
+const getBaseOptions = () => ({
+  clusters: {},
+  allEphemeralIds: {},
+  versions: {},
+  plugins: {},
+});
+
+describe('Get Logstash Stats', () => {
+  const clusterUuids = ['aCluster', 'bCluster', 'cCluster'];
+  let callCluster = sinon.stub();
+
+  beforeEach(() => {
+    callCluster = sinon.stub();
+  });
+
+  describe('fetchLogstashState', () => {
+    const clusterUuid = 'a';
+    const ephemeralIds = ['a', 'b', 'c'];
+    it('should create the logstash state query correctly', async () => {
+      const expected = {
+        bool: {
+          filter: [
+            {
+              terms: {
+                'logstash_state.pipeline.ephemeral_id': ['a', 'b', 'c'],
+              },
+            },
+            {
+              bool: {
+                must: {
+                  term: { type: 'logstash_state' },
+                },
+              },
+            },
+          ],
+        },
+      };
+
+      await fetchLogstashState(callCluster, clusterUuid, ephemeralIds, {} as any);
+      const { args } = callCluster.firstCall;
+      const [api, { body }] = args;
+      expect(api).toEqual('search');
+      expect(body.query).toEqual(expected);
+    });
+
+    it('should set `from: 0, to: 10000` in the query', async () => {
+      await fetchLogstashState(callCluster, clusterUuid, ephemeralIds, {} as any);
+      const { args } = callCluster.firstCall;
+      const [api, { body }] = args;
+      expect(api).toEqual('search');
+      expect(body.from).toEqual(0);
+      expect(body.size).toEqual(10000);
+    });
+
+    it('should set `from: 10000, to: 10000` in the query', async () => {
+      await fetchLogstashState(callCluster, clusterUuid, ephemeralIds, {
+        page: 1,
+      } as any);
+      const { args } = callCluster.firstCall;
+      const [api, { body }] = args;
+
+      expect(api).toEqual('search');
+      expect(body.from).toEqual(10000);
+      expect(body.size).toEqual(10000);
```

Comment:
This page will likely fail due to `index.max_result_window` (default to 10000). You can find some related docs and workarounds in https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html